src/main/java/org/eclipse/yasson/internal/serializer/CharArraySerializer.java

```diff
@@ -0,0 +1,37 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Oracle and/or its affiliates. All rights reserved.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v1.0 and Eclipse Distribution License v. 1.0
+ * which accompanies this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * Contributors:
+ * Roman Grigoriadi
+ ******************************************************************************/
+
+package org.eclipse.yasson.internal.serializer;
+
+import javax.json.bind.serializer.SerializationContext;
+import javax.json.stream.JsonGenerator;
+
+/**
+ * Serializes byte array as JSON array of ints.
+ *
+ * @author Bernd Zeitler
+ */
+public class CharArraySerializer extends AbstractArraySerializer<char[]> {
+
+    protected CharArraySerializer(SerializerBuilder builder) {
+        super(builder);
+    }
+
+    @Override
+    protected void serializeInternal(char[] obj, JsonGenerator generator, SerializationContext ctx) {
+        for (char c : obj) {
+            generator.write("" + c); // write as string since org.glassfish.json.api.JsonGeneratorImpl does not support writing a char
```

Comment:
Lets use `Character.valueOf(c).toString()` here, since that will make use of the JDK's built-in CharacterCache

Reply:
Ok, I applied the suggested change. Thanks.